### PR TITLE
Process existing cloud nodes in CCM

### LINKED ISSUE #58613: Running cloud-controller-manager should remove uninitialized taint
Once cloud-controller-manager is started we should remove the taint. right?

```
                "taints": [
                    {
                        "effect": "NoSchedule",
                        "key": "node.cloudprovider.kubernetes.io/uninitialized",
                        "value": "true"
                    }
```

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

This is a timing issue. If kubelet(s) get started before the CCM is
started, the shared informer event handler does not process them at
all. So we should loop through these before. We run this in a
go wait.Until loop to tolerate errors when listing the nodes and
giving an opportunity for any scripts that may need to setup RBAC
roles etc.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #58613

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by wlan0** [comment]
@dims I'm a little surprised that this issue is presenting itself now. I know Oracle (@jhorwit2) and Microsoft (@jdumars) have been running CCM for a while now, and they would've certainly run into this before.

I guess I'm looking for a reason to understand why this issue is showing up now. Was there a change in the Controller, Informer, or Queuing logic? 

@cheftako @luxas @thockin Do you know?

**Comment by dims** [comment]
I can consistently recreate this issue with local-up-cluster pointed to openstack external cloud provider. Got this tip from @lavalamp on slack and the change works well.

```
If you're not seeing messages that you expected to see, check if they are being sent as updates. I think syncs might be sent as updates.
```

**Comment by jhorwit2** [comment]
> Do you know?

@wlan0 The only thing I can think of, which I haven't had time to test, is that we don't see this because we aren't setting the cloud provider flag on masters so by the time we have workers joining the cluster the CCM is already running. Since @dims is using local cluster up he only has one node and I don't think there is any logic that makes sure the control plane is up before the kubelet starts. 

I think this change is good (we should move to a queue/retry pattern also). @andrewsykim and I have talked about this before. This change won't cause anymore cloud calls since we won't process the node(s) if they don't have the cloud taint.

**Comment by wlan0** [comment]
@jhorwit2 Thanks for the info. This PR LGTM.

**Review by wlan0 — APPROVED** [review]

File: pkg/controller/cloud/node_controller.go [hunk 1]
```diff
@@ -277,6 +280,14 @@ func (cnc *CloudNodeController) MonitorNode() {
 	}
 }
 
+func (cnc *CloudNodeController) UpdateCloudNode(_, newObj interface{}) {
+	if _, ok := newObj.(*v1.Node); !ok {
+		utilruntime.HandleError(fmt.Errorf("unexpected object type: %v", newObj))
+		return
+	}
+	cnc.AddCloudNode(newObj)
```
[the remark this answers is not in the document]
Comment (dims) [reply pkg/controller/cloud/node_controller.go]: @wlan0 logged another PR for checking cloud taint first in AddCloudNode. Will cut down on calls to cloud provider if we don't need to

**Comment by Random-Liu** [comment]
@dims 
I'm not 100% percent sure. But serial test starts failing heavily after this PR is merged. https://k8s-testgrid.appspot.com/google-gce#gci-gce-serial

It seems that some temporary node started in the test is left over, e.g.:
```
Feb 16 04:14:59.821: All nodes should be ready after test, Not ready nodes: ", bootstrap-e2e-minion-group-9890, bootstrap-e2e-minion-group-b24h"
```

**Comment by dims** [comment]
@Random-Liu it's the other one - https://github.com/kubernetes/kubernetes/pull/59968 revert is in progress

**Pull request merged**